tests/testutil/externalservice.go

```diff
@@ -72,6 +73,38 @@ func (es *TestExternalService) GetFQDN() string {
 	return es.ExternalService.Spec.FQDN
 }
 
+// GetKubeIPAddress retrieves the external service IP Address value using the kubernetes clientset
+func (es *TestExternalService) GetKubeFirstIPAddress() string {
+	clientset := GetKubeClient()
+	endpointsList, err := clientset.CoreV1().
+		Endpoints(testingNamespace).
+		List(metav1.ListOptions{FieldSelector: "metadata.name=" + es.ExternalService.Name})
+	if err != nil {
+		es.T.Fatalf(err.Error())
+	}
+	if len(endpointsList.Items) == 0 ||
+		len(endpointsList.Items[0].Subsets) == 0 ||
+		len(endpointsList.Items[0].Subsets[0].Addresses) == 0 {
+		return ""
+	}
+	return endpointsList.Items[0].Subsets[0].Addresses[0].IP
+}
+
+// GetKubeFQDN retrieves the external service FQDN value using the kubernetes clientset
+func (es *TestExternalService) GetKubeFQDN() string {
+	clientset := GetKubeClient()
+	serviceList, err := clientset.CoreV1().
+		Services(testingNamespace).
+		List(metav1.ListOptions{LabelSelector: "rio.cattle.io/service=" + es.ExternalService.Name})
+	if err != nil {
+		es.T.Fatalf(err.Error())
+	}
+	if len(serviceList.Items) == 0 {
```

Comment:
Might make the above into a helper and then have both these methods just call that so it's not duplicated.

Reply:
Those two are different types, one Endpoints and the other Services what would be the preferred way to do that, type checking?

Reply:
ahh, true